cpp/tests/lists/set_operations/difference_distinct_tests.cpp

```diff
@@ -0,0 +1,663 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/lists/set_operations.hpp>
+#include <cudf/lists/sorting.hpp>
+#include <cudf/lists/stream_compaction.hpp>
+
+using float_type = double;
+using namespace cudf::test::iterators;
+
+auto constexpr null{0};  // null at current level
+auto constexpr XXX{0};   // null pushed down from parent level
+auto constexpr neg_NaN      = -std::numeric_limits<float_type>::quiet_NaN();
+auto constexpr neg_Inf      = -std::numeric_limits<float_type>::infinity();
+auto constexpr NaN          = std::numeric_limits<float_type>::quiet_NaN();
+auto constexpr Inf          = std::numeric_limits<float_type>::infinity();
+auto constexpr NULL_EQUAL   = cudf::null_equality::EQUAL;
+auto constexpr NULL_UNEQUAL = cudf::null_equality::UNEQUAL;
+auto constexpr NAN_EQUAL    = cudf::nan_equality::ALL_EQUAL;
+auto constexpr NAN_UNEQUAL  = cudf::nan_equality::UNEQUAL;
+
+using bools_col     = cudf::test::fixed_width_column_wrapper<bool>;
+using int32s_col    = cudf::test::fixed_width_column_wrapper<int32_t>;
+using floats_lists  = cudf::test::lists_column_wrapper<float_type>;
+using strings_lists = cudf::test::lists_column_wrapper<cudf::string_view>;
+using strings_col   = cudf::test::strings_column_wrapper;
+using structs_col   = cudf::test::structs_column_wrapper;
+using lists_cv      = cudf::lists_column_view;
+
+namespace {
+auto set_op_sorted(cudf::column_view const& lhs,
```

Comment:
Name the operation in the function name. (Otherwise the reader has to infer the operation from the file name.) Same for the other test files.
```suggestion
auto set_difference_sorted(cudf::column_view const& lhs,
```